galpy/potential/SCFPotential.py

```diff
@@ -685,6 +714,108 @@ def integrand(xi, costheta):
+           
+           m - mass of particles
+
+           N - size of the Nth dimension of the expansion coefficients
+
+           L - size of the Lth and Mth dimension of the expansion coefficients
+           
+           a - parameter used to shift the basis functions
+
+           radial_order - Number of sample points of the radial integral. If None, radial_order=max(20, N + 3/2L + 1)
+
+           costheta_order - Number of sample points of the costheta integral. If None, If costheta_order=max(20, L + 1)
+
+           phi_order - Number of sample points of the phi integral. If None, If costheta_order=max(20, L + 1)
+
+        OUTPUT:
+
+           (Acos,Asin) - Expansion coefficients for density dens that can be given to SCFPotential.__init__
+
+        HISTORY:
+
+           2020-11-18 - Written - Morgan Bennett
+
+        """ 
+        
+        n = numpy.arange(0,N)
+        l = numpy.arange(0,L)
+        m = numpy.arange(0,L)
+
+        r = numpy.sqrt(pos[0]**2+pos[1]**2+pos[2]**2)
+        phi = numpy.arctan2(pos[1],pos[0])
+        costheta = pos[2]/r
+        '''
+        Plm= numpy.zeros([len(costheta),1,L,L])
+        for i,ll in enumerate(l):
+            for j,mm in enumerate(m):
+                Plm[:,0,j,i]= lpmv(ll,mm,costheta)
+
+        cosmphi= numpy.cos(phi[:,None]*m[None,:])[:,None,None,:]
+        sinmphi= numpy.sin(phi[:,None]*m[None,:])[:,None,None,:]
+        
+        Ylm= (numpy.sqrt((2.*l[:,None]+1)*gamma(l[:,None]-m[None,:]+1)/gamma(l[:,None]+m[None,:]+1))[None,None,:,:]*Plm)[None,:,:,:,:]*numpy.array([cosmphi,sinmphi])
+        Ylm= numpy.nan_to_num(Ylm)
+
+        C= [[gegenbauer(nn,2.*ll+1.5) for ll in l] for nn in n] 
+        Cn= numpy.zeros((1,len(r),N,L,1))
+        for i in range(N):
+            for j in range(L):
+                Cn[0,:,i,j,0]= C[i][j]((r/a-1)/(r/a+1))
+
+        rl= ((r[:,None]/a)**l[None,:])[None,:,None,:,None]
+        r12l1= ((1+(r[:,None]/a))**(2.*l[None,:]+1))[None,:,None,:,None]
+
+        phinlm= -rl/r12l1*Cn*Ylm
+
+        Sum= numpy.sum(mass[None,:,None,None,None]*phinlm,axis=1)
+        Knl= 0.5*n[:,None]*(n[:,None]+4.*l[None,:]+3.)+(l[None,:]+1)*(2.*l[None,:]+1.)
+
+        Inl= (-Knl*4.*numpy.pi/2.**(8.*l[None,:]+6.)*gamma(n[:,None]+4.*l[None,:]+3.)/gamma(n[:,None]+1)/(n[:,None]+2.*l[None,:]+1.5)/gamma(2.*l[None,:]+1.5)**2)[None,:,:,None]
+
+        Anlm= Inl**(-1)*Sum'''
+        Anlm= numpy.zeros([2,L,L,L])
+        for i,nn in enumerate(n):
+            for j,ll in enumerate(l):
+                for k,mm in enumerate(m[:j+1]):
+
+                    Plm= lpmv(mm,ll,costheta)
+
+                    cosmphi= numpy.cos(phi*mm)
+                    sinmphi= numpy.sin(phi*mm)
+
+                    Ylm= (numpy.sqrt((2.*ll+1)*gamma(ll-mm+1)/gamma(ll+mm+1))*Plm)[None,:]*numpy.array([cosmphi,sinmphi])
+                    Ylm= numpy.nan_to_num(Ylm)
+
+                    C= gegenbauer(nn,2.*ll+1.5)
+                    Cn= C((r/a-1)/(r/a+1))
```

Comment:
Likewise, ``eval_ gegenbauer`` is not an astropy-supported ufunc, so the following is necessary.

``Cn = C( u.Quantity((r/a-1)/(r/a+1), copy=False).to_value(u.one) )``

But there's a computational problem if *r* is close to 0. The argument simplifies to

``Cn= C( u.Quantity((a+1)/(a-1), copy=False).to_value(u.one) )``